components/camel-ipfs/src/main/java/org/apache/camel/component/ipfs/IPFSEndpoint.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.ipfs;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import org.apache.camel.Consumer;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.component.ipfs.IPFSConfiguration.IPFSCommand;
+import org.apache.camel.impl.DefaultEndpoint;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+
+import io.nessus.ipfs.IPFSClient;
+
+/**
+ * The camel-xchange component provide access to many bitcoin and altcoin exchanges for trading and accessing market data.
+ */
+@UriEndpoint(firstVersion = "2.23.0", scheme = "ipfs", title = "IPFS", syntax = ":cmd", producerOnly = true, label = "file,ipfs")
```

Comment:
It's would be nice to use host and port parameters in the syntax for better support in cloud (host:port combination ~> service in k8s or openshift)

Reply:
Like so?

```
syntax = ":cmd?host=127.0.0.1&port=5001"
```

Reply:
@tdiesler , sorry, I'm missing that UriPath's parameters are correct and ready for cloud native apps, but syntax could be  'ipfs:host:port:cmd'.
May be remove default values from host and port to prevent confusing.

Reply:
Is the syntax really 'ipfs:host:port:cmd', so that users would say

```
        CamelContext camelctx = new DefaultCamelContext();
        camelctx.addRoutes(new RouteBuilder() {
            @Override
            public void configure() throws Exception {
                from("direct:start").to("ipfs:127.0.0.1:5001:add");
            }
        });
```

or should it be

```
        CamelContext camelctx = new DefaultCamelContext();
        camelctx.addRoutes(new RouteBuilder() {
            @Override
            public void configure() throws Exception {
                from("direct:start").to("ipfs://127.0.0.1:5001/add");
            }
        });
```